### PR TITLE
feat: mmap configuration

### LINKED ISSUE #1649: Make DataAccess properly configurable through YAML
GH fork can run with DataAccess classes configured to run in mmap mode, which uses less memory by not holding the entire graph data but loading only the parts immediately required. This also leads to a certain degree of performance loss, so this is currently an optional feature activated by configuration.

- [x] Introduce a proper config parameter
- [x] keep it behind a feature flag for now
   - it needs to be activated on demand for now.

### PR DESCRIPTION
**Pull request by takb**
### Pull Request Checklist

- [x] 1. I have [**rebased**][rebase] the latest version of the master branch into my feature branch and all conflicts
         have been resolved.
- [x] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the
         [Unreleased] heading.
- [ ] 3. I have documented my code using JDocs tags.
- [ ] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [ ] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [ ] 6. I have created API tests for any new functionality exposed to the API.
- [ ] 7. If changes/additions are made to the ors-config.json file, I have added these to the [ors config documentation][config]
         along with a short description of what it is for, and documented this in the Pull Request (below).
- [x] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [x] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [x] 10. For new features or changes involving building of graphs, I have tested on a larger dataset
          (at least Germany), and the graphs build without problems (i.e. no out-of-memory errors).
- [ ] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the
          importer etc.), I have generated longer distance routes for the affected profiles with different options
          (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end
          points generated from the current live ORS.
          If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [ ] 12. I have written in the Pull Request information about the changes made including their intended usage
          and why the change was needed.
- [ ] 13. For changes touching the API documentation, I have tested that the API playground [renders correctly][api].

Fixes #1649  .

See Issue for details.

### CODE REVIEW

File: ors-api/ors-config.yml [hunk 1]
```diff
@@ -7,13 +7,20 @@
...
+#    allowed_headers: Content-Type, X-Requested-With, accept, Origin, Access-Control-Request-Method, Access-Control-Request-Headers, Authorization
+#    preflight_max_age: 600
   engine:
-    ### use the following line for development setup using test OSM file for Heidelberg
-    #source_file: ./src/test/files/heidelberg.osm.gz
+    ### use the following line to enable mmap mode
+#    graphs_data_access: MMAP
+
+    ### use the following lines for development setup using test OSM file for Heidelberg
+#    source_file: ./src/test/files/heidelberg.osm.gz
```
Comment (aoles) [inline ors-api/ors-config.yml]: Does the path rooted in `./src/...` make sense? It is inconsistent with the uncommented part which uses `./ors-core`. I think the recommendation is to have the working directory set to the project parent dir `openrouteservice`, at least this is how I have it set up locally..

**Review by aoles — CHANGES_REQUESTED** [review]
Thanks! Looks quite good, see my comment for some minor issue I've noticed.

File: ors-api/ors-config.yml (hunk 1 above)
Comment (takb) [reply ors-api/ors-config.yml]: Okay, I'll need to make sure the paths in the docs are good

File: ors-api/ors-config.yml [hunk 2]
```diff
@@ -12,17 +12,17 @@ ors:
 #    allowed_headers: Content-Type, X-Requested-With, accept, Origin, Access-Control-Request-Method, Access-Control-Request-Headers, Authorization
 #    preflight_max_age: 600
   engine:
+    graphs_root_path: ./ors-core/data/graphs
```
Comment (aoles) [inline ors-api/ors-config.yml:15]: Is there any specific reason why the default is to have `ors-core` subdirectory? For example, when ORS is run from the command line the cwd is set to `ors-api`, so the paths become `ors-api/./ors-core/...`. I would suggest to skip on `ors-core` entirely.

Comment (takb) [reply ors-api/ors-config.yml:15]: Yes, it reflects the setup used within the docker container. Personally I think that's something we could get rid of when we redo the docker setup after we merge the jar deployment...

**Review by aoles — APPROVED** [review]
Cheers! 🚀

**Pull request merged**